### PR TITLE
ADEN-4183 Animate VUAP

### PR DESCRIPTION
**Pull request by drets**
Make switching between video ad and image ad more smooth.

### CODE REVIEW

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 1]
```diff
@@ -12,6 +13,7 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js:13]: MAJOR Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

File: extensions/wikia/AdEngine/js/template/bfaaMobile.js [hunk 2]
```diff
@@ -12,6 +13,7 @@ define('ext.wikia.adEngine.template.bfaaMobile', [
 ], function (
```
Comment (WikiaTech) [inline extensions/wikia/AdEngine/js/template/bfaaMobile.js:13]: MAJOR Function has 10 parameters which is greater than 7 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues
- MAJOR 2 major

Watch the comments in this conversation to review them.

File: extensions/wikia/AdEngine/js/template/bfaaDesktop.js [hunk 3]
```diff
@@ -78,41 +83,77 @@ define('ext.wikia.adEngine.template.bfaaDesktop', [
 	}
 
 	function getSlotSize(params) {
+		var width = document.body.clientWidth;
 		return {
-			width: document.body.clientWidth,
-			height: document.body.clientWidth / params.videoAspectRatio
+			width: width,
+			videoHeight: width / params.videoAspectRatio,
+			adHeight: width / params.aspectRatio
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/template/bfaaDesktop.js:90]: :+1:

File: extensions/wikia/AdEngine/js/video/videoAdFactory.js [hunk 4]
```diff
@@ -13,47 +13,32 @@ define('ext.wikia.adEngine.video.videoAdFactory', [
 		return googleIma.init();
 	}
 
-	function create(imageContainer, slotSize, adSlot, slotParams) {
-		var vastUrl = vastUrlBuilder.build(slotSize.width / slotSize.height, slotParams);
+	function create(width, height, adSlot, slotParams) {
```
Comment (v3rth) [inline extensions/wikia/AdEngine/js/video/videoAdFactory.js:16]: Why you choose single value except object?

**Review by v3rth — APPROVED** [review]
LGTM

File: extensions/wikia/AdEngine/js/video/videoAdFactory.js (hunk 4 above)
Comment (drets) [reply extensions/wikia/AdEngine/js/video/videoAdFactory.js:16]: Answered offline.

**Comment by bulaszenko** [comment]
checked works good

**Pull request merged**